packages/library-base/src/commonMain/kotlin/io/realm/internal/Observable.kt

```diff
@@ -21,23 +21,25 @@ import kotlinx.coroutines.channels.ChannelResult
 import kotlinx.coroutines.channels.SendChannel
 import kotlinx.coroutines.flow.Flow
 
-// FIXME Could be split into various interfaces, to getter better type safety around our operations
-//  - Native reference
-//    val ref: NativePointer
-//  - Freezable
-//    fun freeze(frozenRealm: RealmReference): Frozen<T>
-//  - Thawable
-//    fun thaw(liveRealm: RealmReference): Live<T>?
-//    fun registerForNotification(callback: Callback): NativePointer
-//    fun emitFrozenUpdate(frozenRealm: RealmReference, change: NativePointer, channel: SendChannel<T>): ChannelResult<Unit>?
-//  - Public Observable
-//    fun observe(): Flow<T>
 interface Observable<T> {
-    fun freeze(frozenRealm: RealmReference): Observable<T>?
-    fun thaw(liveRealm: RealmReference): Observable<T>?
     fun registerForNotification(callback: Callback): NativePointer
+
     // FIXME Needs elaborate doc on how to signal and close channel
-    fun emitFrozenUpdate(frozenRealm: RealmReference, change: NativePointer, channel: SendChannel<T>): ChannelResult<Unit>?
-    // Should we have a similar public variant
-    fun observe(): Flow<T>
+    fun emitFrozenUpdate(
+        frozenRealm: RealmReference,
+        change: NativePointer,
+        channel: SendChannel<T>
+    ): ChannelResult<Unit>?
+}
+
+interface Freezable<T> {
+    fun freeze(frozenRealm: RealmReference): Observable<T>?
```

Comment:
Maybe mis-phrased a bit and maybe also not critical enough to change, but I will just try again to clarify. I get why you made the split, but if you split out the `Freezable` wouldn't it then be sufficient to return `T` instead of `Observable<T>`:
```
 fun freeze(frozenRealm: RealmReference): T?
```
And then we could catch the connection to `Observable<T>` only in the aforementioned grouping interface? 

Anyway, just thoughts. As said I am ready to go with your `Freezable<T>`